test/extensions/filters/http/proto_api_scrubber/filter_test.cc

```diff
@@ -581,15 +651,284 @@ TEST_F(ProtoApiScrubberScrubbingTest, ScrubRequestNestedField) {
   EXPECT_EQ(Envoy::Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(req_headers, true));
   EXPECT_EQ(Envoy::Http::FilterDataStatus::Continue, filter_->decodeData(*request_data, true));
 
-  // Post-check: Verify scrubbing happened
+  // Post-check: Verify scrubbing happened.
   CreateApiKeyRequest expected_scrubbed_request = makeCreateApiKeyRequest();
   expected_scrubbed_request.mutable_key()->mutable_update_time()->clear_seconds();
 
   checkSerializedData<CreateApiKeyRequest>(*request_data, {expected_scrubbed_request});
 }
 
-} // namespace
+// Tests for Method Level Restrictions.
+class MethodLevelRestrictionTest : public ProtoApiScrubberFilterTest {
+protected:
+  // Override setup to load bookstore descriptor.
+  void setup() override {
+    setupMocks();
+    // Config will be set by each test.
+  }
+};
+
+// Tests that a request is blocked if the method-level matcher evaluates to true.
+TEST_F(MethodLevelRestrictionTest, MethodBlockedByMatcher) {
+  setupFilterConfig(R"pb(
+    restrictions: {
```

Comment:
nit: use yaml instead of this (no need for curly brackets everywhere, and makes the code a bit more concise - [example](https://github.com/envoyproxy/envoy/blob/716a49650f551c6211a196f5828a5a6deaf5080c/test/integration/extension_discovery_integration_test.cc#L27)).